From 64035f573f9b340cf96cd666dc00a1126bc0e184 Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Mon, 11 Nov 2013 16:59:16 -0800
Subject: Dalvik: Vectorization uses lastMIRInsn->next

BZ: 151256

The Vectorization pass uses lastMIRInsn->next to determine the end of a
 traversal. However, there is no reason for that as 0 suffices.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I088bb2579542a07453b9b27b69da666e2f999d1f
Orig-MCG-Change-Id: I118fd1a5cb61b937c918a6efd18c6f62dc5a45c5
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Vectorization.cpp |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/vm/compiler/Vectorization.cpp b/vm/compiler/Vectorization.cpp
index c6f599c..ed8ed88 100644
--- a/vm/compiler/Vectorization.cpp
+++ b/vm/compiler/Vectorization.cpp
@@ -526,9 +526,7 @@ static bool fillVectorizationInformation (const CompilationUnit *cUnit, LoopInfo
     info->constants[increment] = 0;
 
     //Go through the MIRs of the BB and fill up information
-    MIR *mir;
-
-    for (mir = bb->firstMIRInsn; mir != bb->lastMIRInsn->next; mir = mir->next)
+    for (MIR *mir = bb->firstMIRInsn; mir != 0; mir = mir->next)
     {
         //Get the SSA
         SSARepresentation *ssaRep = mir->ssaRep;
-- 
1.7.4.1

